src/OpenLoco/GameCommands/RemoveBuilding.cpp

```diff
@@ -0,0 +1,100 @@
+#include "../Economy/Economy.h"
+#include "../Industry.h"
+#include "../Localisation/FormatArguments.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Map/TileManager.h"
+#include "../Objects/BuildingObject.h"
+#include "../S5/S5.h"
+#include "../TownManager.h"
+#include "GameCommands.h"
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00497D8D
+    static int8_t sub_497D8D(Map::Pos2& pos)
+    {
+        registers regs;
+        regs.ax = pos.x;
+        regs.cx = pos.y;
+        return call(0x00497D8D, regs);
+    }
+
+    // 0x0042D74E
+    static uint32_t removeBuilding(Map::Pos3& pos, uint8_t flags)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Construction);
+        GameCommands::setPosition(pos);
+
+        auto tile = Map::TileManager::get(pos);
+
+        for (auto& el : tile)
+        {
+            if (el.type() != Map::ElementType::building)
+                continue;
+
+            if (el.baseZ() != pos.z / 4)
+                continue;
+
+            auto& elBuilding = el.get<Map::BuildingElement>();
+            const auto* buildingObj = elBuilding.getObject();
+            if (!isEditorMode())
+            {
+                if ((flags & (GameCommands::Flags::flag_6 | GameCommands::Flags::flag_7)) == 0)
+                {
+                    if ((buildingObj->flags & BuildingObjectFlags::undestructible) != 0)
+                    {
+                        GameCommands::setErrorText(StringIds::tooltip_demolition_not_allowed);
+                        return GameCommands::FAILURE;
+                    }
+
+                    if ((buildingObj->flags & BuildingObjectFlags::misc_building) != 0)
+                    {
+                        auto rating = sub_497D8D(pos);
+                        if (rating < 0)
+                        {
+                            auto res = TownManager::getClosestTownAndUnk(pos);
+                            auto* town = TownManager::get(res->first);
+                            auto formatArgs = FormatArguments::common();
+                            formatArgs.push(town->name);
+                            GameCommands::setErrorText(StringIds::local_authority_refuses_permission);
+                            return GameCommands::FAILURE;
+                        }
+                    }
+                }
+            }
+            auto animOffsets = getUnk4F9274(buildingObj->flags & BuildingObjectFlags::large_tile);
+            for (auto animOffset : animOffsets)
```

Comment:
these are sub tile offsets for multi tile buildings. So if the building uses 4 tiles it will iterate through all 4 giving you offsets for each one which you add to the base position to get the buildings element.

Reply:
I added a `sub` prefix to the local variables.